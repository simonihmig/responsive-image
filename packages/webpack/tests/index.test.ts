import { describe, expect, test } from 'vitest';
import { dirname, join } from 'path';
import compiler from './compiler';
import { fileURLToPath } from 'url';
import { configureToMatchImageSnapshot } from 'jest-image-snapshot';

const customConfig = { threshold: 0.1 };
const toMatchImageSnapshot = configureToMatchImageSnapshot({
  customDiffConfig: customConfig,
});

expect.extend({ toMatchImageSnapshot });

const _dirname = dirname(fileURLToPath(import.meta.url));

// make sure we remove any non-deterministic parts from the output
function sanitizeOutput(output: string | Buffer | undefined) {
  return String(output).replace(new RegExp(_dirname, 'g'), '');
}

test('it produces expected output', async () => {
  const { stats, fs } = await compiler(
    'fixtures/image.jpg?responsive',
    _dirname,
    {
      format: ['png', 'webp'],
    },
  );

  expect(stats.modules).toBeDefined();
  expect(stats.modules![0]?.modules).toHaveLength(2);

  const output = stats.modules?.[0]?.modules?.[0]?.source;
  expect(sanitizeOutput(output)).toMatchSnapshot();

  const imageAssets = stats.modules![0]!.assets!;
  expect(imageAssets.toSorted()).toEqual(
    [
      'images/image-640w.png',
      'images/image-640w.webp',
      'images/image-750w.png',
      'images/image-750w.webp',
      'images/image-828w.png',
      'images/image-828w.webp',
      'images/image-1080w.png',
      'images/image-1080w.webp',
      'images/image-1200w.png',
      'images/image-1200w.webp',
      'images/image-1920w.png',
      'images/image-1920w.webp',
      'images/image-2048w.png',
      'images/image-2048w.webp',
      'images/image-3840w.png',
      'images/image-3840w.webp',
    ].toSorted(),
  );

  expect(
    fs.readFileSync(join(_dirname, imageAssets[0] as string)),
  ).toMatchImageSnapshot();
});

test('custom loader options are supported', async () => {
  const { stats, fs } = await compiler(
    'fixtures/image.jpg?responsive',
    _dirname,
    {
      w: [100, 200],
      format: ['png'],
      name: 'test-[width].[ext]',
      webPath: 'https://cdn.example.com/images',
    },
  );

  expect(stats.modules).toBeDefined();
  expect(stats.modules![0]?.modules).toHaveLength(2);

  const output = stats.modules?.[0]?.modules?.[0]?.source;
  expect(sanitizeOutput(output)).toMatchSnapshot();

  const imageAssets = stats.modules![0]!.assets!;
  expect(imageAssets).toEqual(['images/test-100.png', 'images/test-200.png']);

  for (const image of imageAssets) {
    expect(
      fs.readFileSync(join(_dirname, image as string)),
    ).toMatchImageSnapshot();
  }
});

test('custom query params are supported', async () => {
  const { stats, fs } = await compiler(
    'fixtures/image.jpg?w=100;200&format=png&responsive',
    _dirname,
    {},
  );

  expect(stats.modules).toBeDefined();
  // expect(stats.modules![0]?.modules).toHaveLength(3);

  const output = stats.modules?.[0]?.modules?.[0]?.source;
  expect(sanitizeOutput(output)).toMatchSnapshot();

  const imageAssets = stats.modules![0]!.assets!;
  expect(imageAssets).toEqual([
    'images/image-100w.png',
    'images/image-200w.png',
  ]);

  for (const image of imageAssets) {
    expect(
      fs.readFileSync(join(_dirname, image as string)),
    ).toMatchImageSnapshot();
  }
});

test('imagetools params are supported', async () => {
  const { stats, fs } = await compiler(
    'fixtures/image.jpg?w=100;200&format=png&grayscale&responsive',
    _dirname,
    {},
  );

  expect(stats.modules).toBeDefined();

  const output = stats.modules?.[0]?.modules?.[0]?.source;
  expect(sanitizeOutput(output)).toMatchSnapshot();

  const imageAssets = stats.modules![0]!.assets!;
  expect(imageAssets).toEqual([
    'images/image-100w.png',
    'images/image-200w.png',
  ]);

  for (const image of imageAssets) {
    expect(
      fs.readFileSync(join(_dirname, image as string)),
    ).toMatchImageSnapshot();
  }
});

describe('LQIP', function () {
  test('color LQIP is supported', async () => {
    const { stats } = await compiler(
      'fixtures/image.jpg?responsive',
      _dirname,
      {
        lqip: { type: 'color' },
      },
    );

    expect(stats.modules).toBeDefined();
    expect(stats.modules![0]?.modules).toHaveLength(3);

    const output = stats.modules?.[0]?.modules?.[0]?.source;
    expect(sanitizeOutput(output)).toMatchSnapshot();
  });

  test('inline LQIP is supported', async () => {
    const { stats } = await compiler(
      'fixtures/image.jpg?responsive',
      _dirname,
      {
        lqip: { type: 'inline' },
      },
    );

    expect(stats.modules).toBeDefined();
    expect(stats.modules![0]?.modules).toHaveLength(3);

    const output = stats.modules?.[0]?.modules?.[0]?.source;
    expect(sanitizeOutput(output)).toMatchSnapshot();
  });

  test('blurhash LQIP is supported', async () => {
    const { stats } = await compiler(
      'fixtures/image.jpg?responsive',
      _dirname,
      {
        lqip: { type: 'blurhash' },
      },
    );

    expect(stats.modules).toBeDefined();
    expect(stats.modules![0]?.modules).toHaveLength(2);

    const output = stats.modules?.[0]?.modules?.[0]?.source;
    expect(sanitizeOutput(output)).toMatchSnapshot();
  });
});
