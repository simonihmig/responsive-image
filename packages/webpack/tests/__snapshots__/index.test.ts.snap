// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LQIP > blurhash LQIP is supported 1`] = `
"import { findMatchingImage } from '@responsive-image/core';
import { decode2url } from '@responsive-image/core/blurhash/decode';
const images = [{
  url: __webpack_public_path__ + "assets/image-640w.jpg",
  width: 640,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-640w.webp",
  width: 640,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-750w.jpg",
  width: 750,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-750w.webp",
  width: 750,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-828w.jpg",
  width: 828,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-828w.webp",
  width: 828,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-1080w.jpg",
  width: 1080,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-1080w.webp",
  width: 1080,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-1200w.jpg",
  width: 1200,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-1200w.webp",
  width: 1200,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-1920w.jpg",
  width: 1920,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-1920w.webp",
  width: 1920,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-2048w.jpg",
  width: 2048,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-2048w.webp",
  width: 2048,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-3840w.jpg",
  width: 3840,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-3840w.webp",
  width: 3840,
  format: 'webp'
}];
export default {
  imageTypes: [
    'jpeg',
    'webp'
  ],
  availableWidths: [
    640,
    750,
    828,
    1080,
    1200,
    1920,
    2048,
    3840
  ],
  aspectRatio: 1.5,
  imageUrlFor: (w, f) => findMatchingImage(images, w, f ?? 'jpeg')?.url,
  lqip: {
    bgImage: () => decode2url('M53dVIR8D8*Ft2MpoxylRoRlHYniyBRQbu', 5, 3),
    attribute: 'bh:5:3:M53dVIR8D8*Ft2MpoxylRoRlHYniyBRQbu'
  }
}"
`;

exports[`LQIP > color LQIP is supported 1`] = `
"import { findMatchingImage } from '@responsive-image/core';
import '/fixtures/image.jpg.css!=!@responsive-image/webpack/lqip/color-css!/fixtures/image.jpg?className=ri-dyn-0';
const images = [{
  url: __webpack_public_path__ + "assets/image-640w.jpg",
  width: 640,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-640w.webp",
  width: 640,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-750w.jpg",
  width: 750,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-750w.webp",
  width: 750,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-828w.jpg",
  width: 828,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-828w.webp",
  width: 828,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-1080w.jpg",
  width: 1080,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-1080w.webp",
  width: 1080,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-1200w.jpg",
  width: 1200,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-1200w.webp",
  width: 1200,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-1920w.jpg",
  width: 1920,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-1920w.webp",
  width: 1920,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-2048w.jpg",
  width: 2048,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-2048w.webp",
  width: 2048,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-3840w.jpg",
  width: 3840,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-3840w.webp",
  width: 3840,
  format: 'webp'
}];
export default {
  imageTypes: [
    'jpeg',
    'webp'
  ],
  availableWidths: [
    640,
    750,
    828,
    1080,
    1200,
    1920,
    2048,
    3840
  ],
  aspectRatio: 1.5,
  imageUrlFor: (w, f) => findMatchingImage(images, w, f ?? 'jpeg')?.url,
  lqip: {
    'class': 'ri-dyn-0'
  }
}"
`;

exports[`LQIP > color LQIP with inline styles is supported 1`] = `
"import { findMatchingImage } from '@responsive-image/core';
import inlineStyles from '@responsive-image/webpack/lqip/color-css!/fixtures/image.jpg?className=ri-dyn-0&inline=1';
const images = [{
  url: __webpack_public_path__ + "assets/image-640w.jpg",
  width: 640,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-640w.webp",
  width: 640,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-750w.jpg",
  width: 750,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-750w.webp",
  width: 750,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-828w.jpg",
  width: 828,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-828w.webp",
  width: 828,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-1080w.jpg",
  width: 1080,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-1080w.webp",
  width: 1080,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-1200w.jpg",
  width: 1200,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-1200w.webp",
  width: 1200,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-1920w.jpg",
  width: 1920,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-1920w.webp",
  width: 1920,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-2048w.jpg",
  width: 2048,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-2048w.webp",
  width: 2048,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-3840w.jpg",
  width: 3840,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-3840w.webp",
  width: 3840,
  format: 'webp'
}];
export default {
  imageTypes: [
    'jpeg',
    'webp'
  ],
  availableWidths: [
    640,
    750,
    828,
    1080,
    1200,
    1920,
    2048,
    3840
  ],
  aspectRatio: 1.5,
  imageUrlFor: (w, f) => findMatchingImage(images, w, f ?? 'jpeg')?.url,
  lqip: {
    inlineStyles: inlineStyles
  }
}"
`;

exports[`LQIP > inline LQIP is supported 1`] = `
"import { findMatchingImage } from '@responsive-image/core';
import '/fixtures/image.jpg.css!=!@responsive-image/webpack/lqip/inline-css!/fixtures/image.jpg?className=ri-dyn-0&targetPixels=60';
const images = [{
  url: __webpack_public_path__ + "assets/image-640w.jpg",
  width: 640,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-640w.webp",
  width: 640,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-750w.jpg",
  width: 750,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-750w.webp",
  width: 750,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-828w.jpg",
  width: 828,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-828w.webp",
  width: 828,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-1080w.jpg",
  width: 1080,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-1080w.webp",
  width: 1080,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-1200w.jpg",
  width: 1200,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-1200w.webp",
  width: 1200,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-1920w.jpg",
  width: 1920,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-1920w.webp",
  width: 1920,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-2048w.jpg",
  width: 2048,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-2048w.webp",
  width: 2048,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-3840w.jpg",
  width: 3840,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-3840w.webp",
  width: 3840,
  format: 'webp'
}];
export default {
  imageTypes: [
    'jpeg',
    'webp'
  ],
  availableWidths: [
    640,
    750,
    828,
    1080,
    1200,
    1920,
    2048,
    3840
  ],
  aspectRatio: 1.5,
  imageUrlFor: (w, f) => findMatchingImage(images, w, f ?? 'jpeg')?.url,
  lqip: {
    'class': 'ri-dyn-0'
  }
}"
`;

exports[`LQIP > inline LQIP with inline styles is supported 1`] = `
"import { findMatchingImage } from '@responsive-image/core';
import inlineStyles from '@responsive-image/webpack/lqip/inline-css!/fixtures/image.jpg?&targetPixels=60&inline=1';
const images = [{
  url: __webpack_public_path__ + "assets/image-640w.jpg",
  width: 640,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-640w.webp",
  width: 640,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-750w.jpg",
  width: 750,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-750w.webp",
  width: 750,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-828w.jpg",
  width: 828,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-828w.webp",
  width: 828,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-1080w.jpg",
  width: 1080,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-1080w.webp",
  width: 1080,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-1200w.jpg",
  width: 1200,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-1200w.webp",
  width: 1200,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-1920w.jpg",
  width: 1920,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-1920w.webp",
  width: 1920,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-2048w.jpg",
  width: 2048,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-2048w.webp",
  width: 2048,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-3840w.jpg",
  width: 3840,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-3840w.webp",
  width: 3840,
  format: 'webp'
}];
export default {
  imageTypes: [
    'jpeg',
    'webp'
  ],
  availableWidths: [
    640,
    750,
    828,
    1080,
    1200,
    1920,
    2048,
    3840
  ],
  aspectRatio: 1.5,
  imageUrlFor: (w, f) => findMatchingImage(images, w, f ?? 'jpeg')?.url,
  lqip: {
    inlineStyles: inlineStyles
  }
}"
`;

exports[`LQIP > thumbhash LQIP is supported 1`] = `
"import { findMatchingImage } from '@responsive-image/core';
import { decode2url } from '@responsive-image/core/thumbhash/decode';
const images = [{
  url: __webpack_public_path__ + "assets/image-640w.jpg",
  width: 640,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-640w.webp",
  width: 640,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-750w.jpg",
  width: 750,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-750w.webp",
  width: 750,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-828w.jpg",
  width: 828,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-828w.webp",
  width: 828,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-1080w.jpg",
  width: 1080,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-1080w.webp",
  width: 1080,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-1200w.jpg",
  width: 1200,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-1200w.webp",
  width: 1200,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-1920w.jpg",
  width: 1920,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-1920w.webp",
  width: 1920,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-2048w.jpg",
  width: 2048,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-2048w.webp",
  width: 2048,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-3840w.jpg",
  width: 3840,
  format: 'jpeg'
},{
  url: __webpack_public_path__ + "assets/image-3840w.webp",
  width: 3840,
  format: 'webp'
}];
export default {
  imageTypes: [
    'jpeg',
    'webp'
  ],
  availableWidths: [
    640,
    750,
    828,
    1080,
    1200,
    1920,
    2048,
    3840
  ],
  aspectRatio: 1.5,
  imageUrlFor: (w, f) => findMatchingImage(images, w, f ?? 'jpeg')?.url,
  lqip: {
    bgImage: () => decode2url('jJcFFYI1fIWHe4dweXlYeUaAmWj3', 100, 67),
    attribute: 'th:jJcFFYI1fIWHe4dweXlYeUaAmWj3'
  }
}"
`;

exports[`custom loader options are supported 1`] = `
"import { findMatchingImage } from '@responsive-image/core';
const images = [{
  url: '"https:/cdn.example.com/images/test-100.png"',
  width: 100,
  format: 'png'
},{
  url: '"https:/cdn.example.com/images/test-200.png"',
  width: 200,
  format: 'png'
}];
export default {
  imageTypes: [
    'png'
  ],
  availableWidths: [
    100,
    200
  ],
  aspectRatio: 1.5,
  imageUrlFor: (w, f) => findMatchingImage(images, w, f ?? 'png')?.url,
  lqip: undefined
}"
`;

exports[`custom query params are supported 1`] = `
"import { findMatchingImage } from '@responsive-image/core';
const images = [{
  url: __webpack_public_path__ + "assets/image-100w.png",
  width: 100,
  format: 'png'
},{
  url: __webpack_public_path__ + "assets/image-200w.png",
  width: 200,
  format: 'png'
}];
export default {
  imageTypes: [
    'png'
  ],
  availableWidths: [
    100,
    200
  ],
  aspectRatio: 1.5,
  imageUrlFor: (w, f) => findMatchingImage(images, w, f ?? 'png')?.url,
  lqip: undefined
}"
`;

exports[`different aspect ratio 1`] = `
"import { findMatchingImage } from '@responsive-image/core';
const images = [{
  url: __webpack_public_path__ + "assets/image-100w.png",
  width: 100,
  format: 'png'
},{
  url: __webpack_public_path__ + "assets/image-200w.png",
  width: 200,
  format: 'png'
}];
export default {
  imageTypes: [
    'png'
  ],
  availableWidths: [
    100,
    200
  ],
  aspectRatio: 0.6666666666666666,
  imageUrlFor: (w, f) => findMatchingImage(images, w, f ?? 'png')?.url,
  lqip: undefined
}"
`;

exports[`imagetools params are supported 1`] = `
"import { findMatchingImage } from '@responsive-image/core';
const images = [{
  url: __webpack_public_path__ + "assets/image-100w.png",
  width: 100,
  format: 'png'
},{
  url: __webpack_public_path__ + "assets/image-200w.png",
  width: 200,
  format: 'png'
}];
export default {
  imageTypes: [
    'png'
  ],
  availableWidths: [
    100,
    200
  ],
  aspectRatio: 1.5,
  imageUrlFor: (w, f) => findMatchingImage(images, w, f ?? 'png')?.url,
  lqip: undefined
}"
`;

exports[`it produces expected output 1`] = `
"import { findMatchingImage } from '@responsive-image/core';
const images = [{
  url: __webpack_public_path__ + "assets/image-640w.png",
  width: 640,
  format: 'png'
},{
  url: __webpack_public_path__ + "assets/image-640w.webp",
  width: 640,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-750w.png",
  width: 750,
  format: 'png'
},{
  url: __webpack_public_path__ + "assets/image-750w.webp",
  width: 750,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-828w.png",
  width: 828,
  format: 'png'
},{
  url: __webpack_public_path__ + "assets/image-828w.webp",
  width: 828,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-1080w.png",
  width: 1080,
  format: 'png'
},{
  url: __webpack_public_path__ + "assets/image-1080w.webp",
  width: 1080,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-1200w.png",
  width: 1200,
  format: 'png'
},{
  url: __webpack_public_path__ + "assets/image-1200w.webp",
  width: 1200,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-1920w.png",
  width: 1920,
  format: 'png'
},{
  url: __webpack_public_path__ + "assets/image-1920w.webp",
  width: 1920,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-2048w.png",
  width: 2048,
  format: 'png'
},{
  url: __webpack_public_path__ + "assets/image-2048w.webp",
  width: 2048,
  format: 'webp'
},{
  url: __webpack_public_path__ + "assets/image-3840w.png",
  width: 3840,
  format: 'png'
},{
  url: __webpack_public_path__ + "assets/image-3840w.webp",
  width: 3840,
  format: 'webp'
}];
export default {
  imageTypes: [
    'png',
    'webp'
  ],
  availableWidths: [
    640,
    750,
    828,
    1080,
    1200,
    1920,
    2048,
    3840
  ],
  aspectRatio: 1.5,
  imageUrlFor: (w, f) => findMatchingImage(images, w, f ?? 'png')?.url,
  lqip: undefined
}"
`;
